File: src/api/java/com/spicymemes/api/RecipesRegistryManager.java
package com.spicymemes.api;

import java.util.HashMap;

public class RecipesRegistryManager {
    private static HashMap<String, RecipesBase> registries = new HashMap<>();

    private static String createRegistryName(String modId, String registryName) {
        return modId + ":" + registryName;
    }

    /**
     * Adds a recipe registry to the registry manager
     * @param modId Your mod id
     * @param registryName Your registry name
     * @param registry The registry to add
     */
    public static void addRegistry(String modId, String registryName, RecipesBase registry) {
        registries.put(createRegistryName(modId, registryName), registry);
    }

    /**
     * Gets a recipe registry from the registry manager
     * @param modId The mod id of the registry
     * @param registryName The name of the registry
     * @return The registry, or null if it isn't found
     */
    public static RecipesBase getRegistry(String modId, String registryName) {
        String name = createRegistryName(modId, registryName);
        if (registries.containsKey(name)) {
            return registries.get(name);
        }
        return null;
    }
}

File: common/src/api/kotlin/RecipesRegistryManager.kt
package com.spicymemes.api

object RecipesRegistryManager {

    private val registries: MutableMap<String, RecipesBase> = mutableMapOf()

    private fun createRegistryName(modId: String, registryName: String): String = "$modId:$registryName"

    /**
     * Adds a recipe registry to the registry manager
     * @param modId Your mod id
     * @param registryName Your registry name
     * @param registry The registry to add
     */
    @JvmStatic
    fun addRegistry(modId: String, registryName: String, registry: RecipesBase) {
        registries[createRegistryName(modId, registryName)] = registry
    }

    /**
     * Gets a recipe registry from the registry manager
     * @param modId The mod id of the registry
     * @param registryName The name of the registry
     * @return The registry, or null if it isn't found
     */
    @JvmStatic
    fun getRegistry(modId: String, registryName: String) =
        registries[createRegistryName(modId, registryName)]
}
